src/main/java/org/candlepin/model/Pool.java

```diff
@@ -864,24 +871,11 @@ public Product getProduct() {
      */
     public Pool setProduct(Product product) {
         this.product = product;
-        this.productUuid = product != null ? product.getUuid() : null;
+        this.productUuid = null;
```

Comment:
While this is correct behavior imo, I wonder if this will trigger the exact warning we're aiming to address.

Reply:
Yeah, I would expect to get that warning as well. However, I ran the changes against the RefreshPoolsSpecTest tests and verified we are hitting the setProduct method, but we are not getting the warning in the logs. I don't fully understand why that is.

Reply:
ack